### ISSUE TITLE
[Bug] How to get the logits or output_log_probs?

### ISSUE THREAD

**Issue by AatroxZhou** [InternLM/lmdeploy#323]
### Checklist

- [ ] 1. I have searched related issues but cannot get the expected help.
- [ ] 2. The bug has not been fixed in the latest version.

### Describe the bug

How to get the logits or output_log_probs?
I add is_return_logits and is_return_log_probs in inputs where in TurbomindInstance, But the outputs have just only the inputs logits and the output_log_probs is all 0 tensor.

### Reproduction

 inputs = dict(
            input_ids=input_ids,
            input_lengths=input_lengths,
            request_output_len=np.full(input_lengths.shape,
                                       request_output_len,
                                       dtype=np.uint32),
            runtime_top_k=_broadcast_np(top_k, np.uint32),
            runtime_top_p=_broadcast_np(top_p, np.float32),
            temperature=_broadcast_np(temperature, np.float32),
            repetition_penalty=_broadcast_np(repetition_penalty, np.float32),
            step=step,
            **is_return_logits=_broadcast_np(1, np.uint32),
            is_return_log_probs=_broadcast_np(1, np.uint32),**
            # session input
            session_len=self.session_len * np.ones([batch_size, ], dtype=np.uint32),
            START=_broadcast_np((1 if sequence_start else 0), np.int32),
            END=_broadcast_np((1 if sequence_end else 0), np.int32),
            CORRID=np.array(session_id, dtype=np.uint64),
            STOP=_broadcast_np((1 if stop else 0), np.int32))

the outputs logits is [batchsize, input_length, vocab_size] 
the outputs output_log_probs is [batchsize, beamwidth, output_length], the value is [[[0.,0.,0.,0.,..........]]]

**Comment by lvhan028** [comment]
Please refer to `lmdeploy/turbomind/decode.py`

**Comment by lzhangzz** [comment]
Currently the generation interface does not support returning logits nor output_log_probs.

One possible solution is to feed the generated tokens into the [decode ](lmdeploy/turbomind/decode.py) interface to get the logits.

**Comment by AatroxZhou** [comment]
> 

Thank you for your reply.
I checked the official meta implementation and found some differences between the two:
1、The way in **decode.py** doesn't seem to receive parameters like top_p,top_k, I tested different top_p,top_k inputs but they don't work. 
2、After decoding top_1 you will find the output is strange, it is non-normal output. Is this normal?

**Comment by AatroxZhou** [comment]
> Currently the generation interface does not support returning logits nor output_log_probs.
> 
> One possible solution is to feed the generated tokens into the [decode ](lmdeploy/turbomind/decode.py) interface to get the logits.

> Please refer to `lmdeploy/turbomind/decode.py`


Thank you for your reply.
I checked the official meta implementation and found some differences between the two:
1、The way in **decode.py** doesn't seem to receive parameters like top_p,top_k, I tested different top_p,top_k inputs but they don't work. 
2、After decoding top_1 you will find the output is strange, it is non-normal output. Is this normal?

**Comment by lvhan028** [comment]
Are you looking for output_log_probs for each generated token?
